runtime/pkg/rduckdb/db.go

```diff
@@ -183,15 +182,17 @@ func NewDB(ctx context.Context, opts *DBOptions) (DB, error) {
 
 	bgctx, cancel := context.WithCancel(context.Background())
 	db := &db{
-		opts:       opts,
-		localPath:  opts.LocalPath,
-		remote:     opts.Remote,
-		writeSem:   semaphore.NewWeighted(1),
-		metaSem:    semaphore.NewWeighted(1),
-		localDirty: true,
-		logger:     opts.Logger,
-		ctx:        bgctx,
-		cancel:     cancel,
+		opts:             opts,
+		localPath:        opts.LocalPath,
+		remote:           opts.Remote,
+		writeSem:         semaphore.NewWeighted(1),
+		metaSem:          semaphore.NewWeighted(1),
+		localDirty:       true,
+		writesInProgress: make(map[string]any),
+		progressMu:       sync.Mutex{},
```

Comment:
Since the number of members is growing big, consider naming it `writesInProgressMu` since it specifically relates to `writesInProgress`.